Modules/_json.c

```diff
@@ -1549,14 +1575,25 @@ encoder_listencode_dict(PyEncoderObject *s, _PyUnicodeWriter *writer,
     if (_PyUnicodeWriter_WriteChar(writer, '{'))
         goto bail;
 
+    PyObject* newline_indent = 0;
+    PyObject* current_item_separator = s->item_separator;
     if (s->indent != Py_None) {
-        /* TODO: DOES NOT RUN */
         indent_level += 1;
-        /*
-            newline_indent = '\n' + (' ' * (_indent * _current_indent_level))
-            separator = _item_separator + newline_indent
-            buf += newline_indent
-        */
+        newline_indent = _create_newline_indent(s->indent, indent_level);
+        if (newline_indent == 0) {
+                goto bail;
```

Comment:
```suggestion
             goto bail;
```
Overindented, hope I counted right. The indentation is also off in a few other lines below.